source/common/access_log/access_log_formatter.h

```diff
@@ -60,10 +60,17 @@ class AccessLogFormatParser {
                            std::vector<std::string>& sub_items, absl::optional<size_t>& max_length);
 
   // the indexes of where the parameters for each directive is expected to begin
+#ifdef _MSC_VER
+  static const size_t ReqParamStart{sizeof("REQ(") - 1};
```

Comment:
This might be tricky. If we declare the strings as `static constexpr req[] = "REQ("`, then compilation fails on Linux with: `error: 'strlen(((const char*)(& Envoy::AccessLog::AccessLogFormatParser::req)))' is not a constant expression`

Reply:
Maybe just have them both do sizeof then and keep a single set of code? It doesn't seem a big deal either way.

Reply:
+1 if possible. I'm a little sketched out each time we have `#ifdef` make its way into code that is not inherently platform specific or some utility header that hides the compiler differences.

Reply:
Sounds good to me